docs/rfc/0696-p2p-node.adoc

```diff
@@ -0,0 +1,318 @@
+be addressed to enable the development of a healthy ecosystem.
+
+== Overview
+
+This RFC expands on the peer-to-peer node outlined in RFC 682 to propose a new
+stand-alone daemon with focus on driving the p2p stack, exposing a minimal API
+and defer supervision to more integrated and mature system services. Furthermore
+it tries to prove that all daemons are made equal and only distinct with regards
+to the configuration given (e.g. tracking behaviour) and lifetime of the process
+(i.e. socket-activated or long-running).
+
+It also strives to recommend APIs and mechanisms - in the form of non-core
+protocols - which remove the need for online requests made by clients to a
+running node.
+
+== Terminology and Conventions
+
+The key words "`MUST`", "`MUST NOT`", "`REQUIRED`", "`SHALL`", "`SHALL NOT`",
+"`SHOULD`", "`SHOULD NOT`", "`RECOMMENDED`", "`NOT RECOMMENDED`", "`MAY`", and
+"`OPTIONAL`" in this document are to be interpreted as described in <<RFC2119>>
+and <<RFC8174>> when, and only when, they appear in all capitals, as shown here.
+
+CBOR <<RFC8949>> datatype definitions are given using the notation devised in
+CDDL <<RFC8610>>. By convention, `struct`-like datatypes are encoded as CBOR
+maps, where the map key is the zero-based numeric index of the field in
+declaration order.
+
+== Protocols
+
+The main purpose of the node is to operate the core protocol of link currently
+implemented in `librad::net::peer`, which currently consists of holding onto the
+futures du jour and dropping them at opportune moments. Furthermore all
+information in form of synchrounous calls and events obtainable internally via
+the sufficiently space-engineered tincans SHALL be made available.
+
+=== Subroutines
+
+This section expands on the auxilliary routines driven by the node on top of the
+core protocol. Where each SHOULD allow to be disabled through configuration.
+
+==== Public API
+
+For other process to be able to talk to the node, it SHALL keep an API
+subroutine running, which offers introspection into its state. See the dedicated
+section for the surface the node will expose.
+
+==== Discovery
+
+In order to guarantee initial connectivity the protocol needs to be given a set
+of bootstrap nodes, while the main means of discovery is part of the membership
+algorithms. Beyond the initial list passed at start-up as source for discovery
+a streaming variant should be - on opt-in basis - made available. Consult the
+<<API>> section for the internals of how that can be achieved by another process.
+
+==== Announcements
+
+Until superseded by graftier ways, currently the main means of dissemination is
+through gossiping new interesting `Have`s. This is achieved by maintainig a
+persisted view on the list of identities and their current refs. During
+periodic runs a delta is created which serves as the basis of the announcements.
+In turn that list is made known to actively connected peers via the core
+protocols gossip.
+
+==== Requests
```

Comment:
Perhaps we could call this `Replication` since its main concern is attempting to replicating an `Urn`? Or `Replication Requests` if we want to distinguish it from bare replication.

Reply:
Disambiguated in 6233e75.